src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -223,8 +223,7 @@ public function supportsNormalization($data, $format = null, array $context = []
      */
     public function supportsDenormalization($data, $type, $format = null, array $context = [])
     {
-        return isset(self::SCALAR_TYPES[$type])
-            || null !== $this->getDenormalizer($data, $type, $format, $context);
+        return isset(self::SCALAR_TYPES[$type]) || null !== $this->getDenormalizer($data, $type, $format, $context);
```

Comment:
shouldn't this be `(isset(self::SCALAR_TYPES[$type]) && is_scalar($data)) || ...`?
worth a test case maybe?

Reply:
When we are deserializing int array with ArrayDenormalizer `$data` is array and `$type = 'int'`, so check for scalar data in that place not possible. But I added this check to `denormalize()` and removed throw because all possible scalar values (as I know) can be casted to int/float/bool/string.